### PR TITLE
create PostComment functionality, updates comment list on screen and …

### PR DESCRIPTION
**Pull request by SandorGonczi**
…gives feedback when new comment is posted

### CODE REVIEW

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
Hi Sandor, functionally your add comment does work. 

I would like you to consider the difference between an input box and a text area. An input box is for one line of text. We would expect comments to be able to add lots of text.

Your new comments are also posting to the bottom of the list. When using sites along the internet it is convention that new comments are at the top of the listing.

Consider these changes and then resubmit please.

File: src/components/Comments.jsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const Comments = ({ article }) => {
         setIsLoading(false);
         setIsError(true);
       });
-  }, [article.article_id]);
+  }, [article.article_id, newComment]);
```
Comment (douglashellowell) [inline src/components/Comments.jsx:21]: ⚠️ watch out!

Whenever `newComment` updates this useEffect will run

This means **every time a user types in the comment box a new request will be made** 😱

File: src/components/Comments.jsx [hunk 2]
```diff
@@ -27,13 +27,21 @@ const Comments = ({ article }) => {
...
-          })}
+          {comments
+            .sort((a, b) =>
+              a.created_at > b.created_at
+                ? -1
+                : a.created_at < b.created_at
+                ? 1
+                : 0
+            )
+            .map((comment, index) => {
```
Comment (douglashellowell) [inline src/components/Comments.jsx:38]: does your backend support a sort_by query?

It's not ideal to do data sorting in the front end

**Comment by douglashellowell** [comment]
You'll need to fix that `useEffect` issue mentioned above - once that's fixed it's ready to merge 👍🏻

File: src/components/Comments.jsx (hunk 1 above)
Comment (SandorGonczi) [reply src/components/Comments.jsx:21]: I will have to refactor that, completly overlooked, it is actually making a request at every character change

File: src/components/Comments.jsx (hunk 2 above)
Comment (SandorGonczi) [reply src/components/Comments.jsx:38]: Unfortunately sort_by is not supported atm when fetching the comments on that endpoint.
It should be added, so there would not be sorting needed.

**Pull request merged**